pandas/tests/base/test_ops.py

```diff
@@ -719,58 +720,63 @@ def test_drop_duplicates_series_vs_dataframe(self):
                 dropped_series = df[column].drop_duplicates(keep=keep)
                 tm.assert_frame_equal(dropped_frame, dropped_series.to_frame())
 
-    def test_fillna(self):
+    def test_fillna(self, index_or_series_obj):
         # # GH 11343
         # though Index.fillna and Series.fillna has separate impl,
         # test here to confirm these works as the same
 
-        for orig in self.objs:
-
-            o = orig.copy()
-            values = o.values
-
-            # values will not be changed
-            result = o.fillna(o.astype(object).values[0])
-            if isinstance(o, Index):
-                tm.assert_index_equal(o, result)
-            else:
-                tm.assert_series_equal(o, result)
-            # check shallow_copied
-            assert o is not result
+        obj = index_or_series_obj
+        if len(obj) == 0:
+            pytest.skip("Test doesn't make sense on empty data")
+        elif isinstance(obj, ABCMultiIndex):
+            pytest.skip("MultiIndex doesn't support isna")
 
-        for null_obj in [np.nan, None]:
-            for orig in self.objs:
-                o = orig.copy()
-                klass = type(o)
+        # values will not be changed
+        result = obj.fillna(obj.values[0])
+        if isinstance(obj, Index):
+            tm.assert_index_equal(obj, result)
+        else:
+            tm.assert_series_equal(obj, result)
 
-                if not allow_na_ops(o):
-                    continue
+        # check shallow_copied
+        assert obj is not result
 
-                if needs_i8_conversion(o):
+    @pytest.mark.parametrize("null_obj", [np.nan, None])
+    def test_fillna_null(self, null_obj, index_or_series_obj):
+        # # GH 11343
+        # though Index.fillna and Series.fillna has separate impl,
+        # test here to confirm these works as the same
+        obj = index_or_series_obj
+        klass = type(obj)
 
-                    values = o.astype(object).values
-                    fill_value = values[0]
-                    values[0:2] = pd.NaT
-                else:
-                    values = o.values.copy()
-                    fill_value = o.values[0]
-                    values[0:2] = null_obj
+        if not allow_na_ops(obj):
+            pytest.skip(f"{klass} doesn't allow for NA operations")
+        elif len(obj) < 1:
+            pytest.skip("Test doesn't make sense on empty data")
+        elif isinstance(obj, ABCMultiIndex):
+            pytest.skip(f"MultiIndex can't hold '{null_obj}'")
```

Comment:
Another open issue?

Reply:
Isn't this by design? A MultiIndex must hold values on multiple levels, so its elements are tuples. Therefore a `MultiIndex` can't just hold a regular `np.nan` or `None` as a value. Maybe a tuple of nans might make sense, but I don't think that would be recognised as "na" by fillna.